src/algebra/group_power/basic.lean

```diff
@@ -432,11 +427,23 @@ alias sub_sq ← sub_pow_two
 lemma sub_sq' (a b : R) : (a - b) ^ 2 = a ^ 2 + b ^ 2 - 2 * a * b :=
 by rw [sub_eq_add_neg, add_sq', neg_sq, mul_neg, ← sub_eq_add_neg]
 
+variables [no_zero_divisors R] {a b : R}
+
+lemma sq_eq_sq_iff_eq_or_eq_neg : a ^ 2 = b ^ 2 ↔ a = b ∨ a = -b :=
+by rw [←sub_eq_zero, sq_sub_sq, mul_eq_zero, add_eq_zero_iff_eq_neg, sub_eq_zero, or_comm]
+
+lemma eq_or_eq_neg_of_sq_eq_sq (a b : R) : a ^ 2 = b ^ 2 → a = b ∨ a = -b :=
+sq_eq_sq_iff_eq_or_eq_neg.1
+
+@[simp] lemma sq_eq_one_iff : a^2 = 1 ↔ a = 1 ∨ a = -1 :=
```

Comment:
As was pointed out by @adomani on Zulip, one does not even need associativity (no triple products involved) or commutativity (`a` commutes with itself, `1` and `-1`) for this (but then, of course, this needs a separate proof, since it is no longer a special case of `sq_eq_sq_iff_eq_or_eq_neg`).

Reply:
If you prove stuff using a `commute` assumption, you can actually not duplicate the proofs.